### ISSUE TITLE
[vue] Rendering crashes `Error: Passing props on "template"!` when having multiple roots on `MenuItem` rendered as `template`

### ISSUE THREAD

**Issue by SimonSimCity** [tailwindlabs/headlessui#2512]
**What package within Headless UI are you using?**

`@headlessui/vue`

**What version of that package are you using?**

`v1.7.13`

**What browser are you using?**

`Chrome` but should happen on any browser...

**Reproduction URL**

https://github.com/SimonSimCity/headlessui-vue-multiple-root-bug

Please go through the history of the project, where I've tried to be as verbose as necessary. The menu works until the last commit, where I added a second root element to the default slot of one of the `MenuItem` components.

**Describe your issue**

On my project, having a `MenuItem` component with multiple root elements, the menu doesn't open but throws an error logged on the console when I try opening the menu:
![Screenshot 2023-05-23 at 16 40 23](https://github.com/tailwindlabs/headlessui/assets/392589/1708dbf7-d565-41d3-9c23-7f7f74df3e73)

It does work as expected when I change the code as following:
* Either, I wrap everything of the slot into another component, so the slot itself just has one root...
* Or I let the component render as a particular html element instead as `template` by setting the `as` property to e.g. `span`.

For this reason it seems that vue-elements, rendered as template, do not support multiple root elements, at least not `MenuItem`.

If this is a known limitation, I have yet failed to see a warning about this in the documentation.

**Comment by thecrypticace** [comment]
Hey! So this is an intentional behavior. It works the same way as a component that renders multiple root nodes in Vue and you try to pass down attrs to be automatically inherited (e.g. `class`).

The reason we can't pass these on to `<template>` with more than one Node is that we don't know which node to apply it to. The answer could very well be "the first one" or "the last one" or, if there are more than two, any one of them.

As the error message says, the only way to fix this is to:
1. Use the `as` prop to wrap the component in a `div` (or some other element)
2. Render a node / component that results in a single HTML element

Given that the error message details the steps necessary to fix the problem (the error is one that we throw inside Headless UI), I'm not certain we need to add this to the docs at this time.

Hope that helps! ✨

**Comment by SimonSimCity** [comment]
Thanks for the explanation - I was quite buffed when I saw this error message and didn't quite know how to deal with it ...

> [...] you try to pass down attrs to be automatically inherited (e.g. `class`).

While this is true in the repository I mentioned this issue, it doesn't apply to the repository I provided for reproduction.

I suggest that this behaviour should be changed to be more in line with how vue handles these cases: https://vuejs.org/guide/components/attrs.html#attribute-inheritance-on-multiple-root-nodes

Vue does not throw an error, but a runtime warning, and I guess it internally rejects the unused data. IMHO, this would also be more intuitive in this case. If you don't know how to handle data, rather disregard it than stopping the show, and let the user know (by logging a warning) that data he provided has been ignored. If there is no data (like in the repo I provided for reproduction), it shouldn't even log out a warning.

Looking forward to hear from you again ✨
